core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/handler/AwsClientHandlerUtils.java

```diff
@@ -100,6 +101,17 @@ private AwsClientHandlerUtils() {
             .putAttribute(SdkInternalExecutionAttribute.DISABLE_HOST_PREFIX_INJECTION,
                           clientConfig.option(SdkAdvancedClientOption.DISABLE_HOST_PREFIX_INJECTION));
 
+        ExecutionAttributes clientOverrideExecutionAttributes = clientConfig.option(SdkClientOption.EXECUTION_ATTRIBUTES);
+        if (clientOverrideExecutionAttributes != null) {
+            executionAttributes = clientOverrideExecutionAttributes.merge(executionAttributes);
+        }
+
+        Optional<RequestOverrideConfiguration> requestOverrideConfiguration =
+                (Optional<RequestOverrideConfiguration>) originalRequest.overrideConfiguration();
```

Comment:
Hmm I don't like the cast. Can we just keep the type as `Optional<? extends RequestOverrideConfiguration>`?